idea/src/org/jetbrains/kotlin/idea/quickfix/AssignOperatorAmbiguityFactory.kt

```diff
@@ -0,0 +1,91 @@
+import org.jetbrains.kotlin.idea.caches.resolve.analyze
+import org.jetbrains.kotlin.idea.references.mainReference
+import org.jetbrains.kotlin.lexer.KtTokens
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.psiUtil.endOffset
+import org.jetbrains.kotlin.resolve.calls.callUtil.getType
+import org.jetbrains.kotlin.resolve.descriptorUtil.isSubclassOf
+import org.jetbrains.kotlin.resolve.lazy.BodyResolveMode
+import org.jetbrains.kotlin.types.KotlinType
+
+object AssignOperatorAmbiguityFactory : KotlinIntentionActionsFactory() {
+    override fun doCreateActions(diagnostic: Diagnostic): List<IntentionAction> {
+        val fixes = mutableListOf<IntentionAction>()
+        val element = diagnostic.psiElement.parent
+        if (element is KtBinaryExpression) {
+            val left = element.left
+            val right = element.right
+            val operationText = when (element.operationToken) {
+                KtTokens.PLUSEQ -> "plus"
+                KtTokens.MINUSEQ -> "minus"
+                else -> null
+            }
+            if (left != null && right != null && operationText != null) {
+                val context = element.analyze(BodyResolveMode.PARTIAL)
+                if (left.getType(context).isMutableCollection()) {
+                    val property = left.mainReference?.resolve() as? KtProperty
+                    val propertyName = property?.name
+                    if (property != null && propertyName != null && property.isLocal) fixes.add(ChangeToValFix(property, propertyName))
+                    fixes.add(ChangeToAssignFunctionCallFix(element, operationText))
+                }
+            }
+        }
+        return fixes
+    }
+}
+
+private fun KotlinType?.isMutableCollection(): Boolean {
+    val classDescriptor = this?.constructor?.declarationDescriptor as? ClassDescriptor ?: return false
+    val className = classDescriptor.name.asString()
+    val builtIns = DefaultBuiltIns.Instance
+    return className.endsWith("List") && classDescriptor.isSubclassOf(builtIns.mutableList)
+            || className.endsWith("Set") && classDescriptor.isSubclassOf(builtIns.mutableSet)
+            || className.endsWith("Map") && classDescriptor.isSubclassOf(builtIns.mutableMap)
+}
+
+private class ChangeToValFix(element: KtProperty, val propertyName: String) : KotlinQuickFixAction<KtProperty>(element) {
+    override fun getText() = "Change '$propertyName' to val"
+
+    override fun getFamilyName() = text
+
+    override fun invoke(project: Project, editor: Editor?, file: KtFile) {
+        val property = element ?: return
+        property.valOrVarKeyword.replace(KtPsiFactory(property).createValKeyword())
+    }
+}
+
+private class ChangeToAssignFunctionCallFix(
+    element: KtBinaryExpression,
+    private val operationText: String
+) : KotlinQuickFixAction<KtBinaryExpression>(element) {
+    override fun getText() = "Change to '${operationText}Assign()' call"
```

Comment:
I'd suggest to change the name to 'replace with ...' as in ReplaceCallFix-quickfixes